### PR TITLE
Host url auto save

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Host URL auto saved

Fixes [AB#688](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/Devtron%20Team/Issues/?workitem=688) 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/create/CreateApp.tsx [hunk 1]
```diff
@@ -56,6 +57,7 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
                 appName: false,
                 cloneAppId: true,
             },
+            hostURLConfig: undefined
```
Comment (vivek-devtron) [inline src/components/app/create/CreateApp.tsx:60]: We should not initialize something with 'undefined' manually

File: src/components/app/create/CreateApp.tsx [hunk 2]
```diff
@@ -75,6 +77,7 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
         }
     }
 
+
```
Comment (vivek-devtron) [inline src/components/app/create/CreateApp.tsx:80]: Please remove extra whitespace

File: src/components/app/create/CreateApp.tsx [hunk 3]
```diff
@@ -139,6 +142,18 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
         return true;
     };
 
+     getHostURLConfig = async() => {
+      const { result } = await getHostURLConfiguration()
```
Comment (vivek-devtron) [inline src/components/app/create/CreateApp.tsx:146]: What will happen in case of error?

File: src/components/hostURL/HostURL.tsx [hunk 4]
```diff
@@ -29,24 +29,44 @@ export default class HostURLConfiguration extends Component<HostURLConfigProps,
...
-            })
-        }).catch((error) => {
-            showError(error);
-            this.setState({ view: ViewType.ERROR, statusCode: error.code });
-        })
+      let payload = {
+          ...this.state.form,
+          value: window.location.origin,
+      }
+      saveHostURLConfiguration(payload)
```
Comment (vivek-devtron) [inline src/components/hostURL/HostURL.tsx:36]: If we are saving the host config and getting form in response, why do we need to call getHostURLConfiguration explicitly. can't we set the same value?

File: src/components/app/create/CreateApp.tsx (hunk 3 above)
Comment (sohel-rp) [reply src/components/app/create/CreateApp.tsx:146]: Wrap in try-catch

File: src/components/app/create/CreateApp.tsx [hunk 5]
```diff
@@ -139,6 +142,18 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
...
+      this.setState({hostURLConfig: result})
+      let payload = {
```
Comment (sohel-rp) [inline src/components/app/create/CreateApp.tsx:148]: Make it const

File: src/components/hostURL/HostURL.tsx [hunk 6]
```diff
@@ -29,24 +29,44 @@ export default class HostURLConfiguration extends Component<HostURLConfigProps,
...
-            }
-            this.setState({
-                view: ViewType.FORM,
-                form: form
-            })
-        }).catch((error) => {
-            showError(error);
-            this.setState({ view: ViewType.ERROR, statusCode: error.code });
-        })
+      let payload = {
```
Comment (sohel-rp) [inline src/components/hostURL/HostURL.tsx:32]: Make it const

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**